mibidata/mibi_image.py

```diff
@@ -199,6 +214,37 @@ def _set_channels(self, channels, length):
                 'Channels must be a list of tuples of (int, str) or a '
                 'list of str')
 
+    def _convert_from_previous_metadata_versions(self):
+        """Convert old metadata format to new one.
+
+        This function ensures backwards compatibility for previous versions.
+        """
+        if self.version != MIBITIFF_VERSION:
+            print("WARNING! you are trying to use the old metadata "
+                  "format for setting the point name. If you are "
+                  "creating a new image, we recommend storing the "
+                  "point name as in the following example: "
+                  "'fov_id': 'Point1', 'fov_name': 'R1C3_Tonsil'.")
+            self.fov_name = self.__dict__.pop('point_name')
+            self.fov_id = self.folder.split('/')[0]
+            self.version = MIBITIFF_VERSION
+
+    def _check_fov_id(self):
+        """Check that fov_id matches folder."""
+        if self.fov_id is None and self.folder is None:
+            pass
+        elif self.fov_id is not None and self.folder is not None:
+            try:
+                assert self.fov_id == self.folder.split('/')[0]
+            except AssertionError:
+                message = (f'The fov_id {self.fov_id} does not '
+                           f'match the folder {self.folder}.')
+                raise ValueError(message)
+        else:
+            message = (f'The fov_id {self.fov_id} does not '
+                       f'match the folder {self.folder}.')
+            raise ValueError(message)
```

Comment:
I think this can be simplified to:
```
if self.fov_id is None and self.folder is None:
  return
try:
  assert self.fov_id == self.folder.split('/')[0]
except (AttributeError, AssertionError):
  raise ValueError(...)

Reply:
Done, the way I had it before was because there is a case in which fov_id can be None but folder not, which is for the old metadata format (using point_name), but I reorganized the check of the old metadata to happen before the check on the new convention, so now it works with the simplified logic.